.github/workflows/build-and-push.yml

```diff
@@ -54,11 +55,20 @@ jobs:
       - name: Set Docker image tag to "latest" for updates of the main branch
         if: github.ref == 'refs/heads/main'
         run: |
-          echo "IMAGE_TAG=latest" >> "$GITHUB_ENV"
+          echo IMAGE_TAG=latest >> "$GITHUB_ENV"
+          # Updates to the main branch are deployed to stage.
+          echo DEPLOYMENT_ENV=stage >> "$GITHUB_ENV"
       - name: Set Docker image tag to the git tag for tagged builds
         if: startsWith(github.ref, 'refs/tags/')
         run: |
-          echo "IMAGE_TAG=$GITHUB_REF_NAME" >> "$GITHUB_ENV"
+          echo IMAGE_TAG="$GITHUB_REF_NAME" >> "$GITHUB_ENV"
+          if [[ $GITHUB_REF_NAME =~ ^v[0-9]{4}.[0-9]{2}.[0-9]{2} ]]; then
+              # Version tags are deployed to prod.
+              echo DEPLOYMENT_ENV=prod >> "$GITHUB_ENV"
+          else
+              # Other tags are not deployed at all.
+              echo DEPLOYMENT_ENV="" >> "$GITHUB_ENV"
```

Comment:
For tags that don't match our `vYYYY.MM.DD[-N]` scheme, we don't want to deploy them. Do we also want to skip pushing them to GAR? If so, should we truncate `IMAGE_TAG` here as well?

I can't think of a case where we'd want to push an image to GAR, but not kick off a deploy. Is that a useful thing to be able to do?

Reply:
I thought we might want to tag some version and then manually deploy it to stage, but that probably never happens. In that case, we probably shouldn't run CI on random tags in the first place, but only on version tags. I'll change this file accordingly and ask for another review.